sys/include/net/cord/lc.h

```diff
@@ -0,0 +1,181 @@
+ */
+enum {
+    CORD_LC_OK        =  0,     /**< everything went as expected */
+    CORD_LC_TIMEOUT   = -1,     /**< no response from the network */
+    CORD_LC_ERR       = -2,     /**< internal error or invalid reply */
+    CORD_LC_OVERFLOW  = -3,     /**< internal buffers can not handle input */
+    CORD_LC_NORSC     = -4,     /**< lookup interface for not found */
+};
+
+#define CORD_LC_ATTR_LEN        (NANOCOAP_URI_MAX)
+
+/**
+ * @brief  RD lookup types
+ */
+enum {
+    CORD_LC_RES,                /**< Resource lookup type */
+    CORD_LC_EP,                 /**< Endpoint lookup type */
+};
+
+/**
+ * @brief  Information about RD server and its lookup interface resources
+ */
+typedef struct {
+    const sock_udp_ep_t *remote;  /**< Remote endpoint of RD server */
+    char res_lookif[CORD_LC_ATTR_LEN];             /**< Location of resource lookup interface */
+    char ep_lookif[CORD_LC_ATTR_LEN];              /**< Location of endpoint lookup interface */
+} cord_lc_rd_t;
+
+/**
+ * @brief  Information about an endpoint registered at a RD server
+ */
+typedef struct {
+    char path[NANOCOAP_URI_MAX];   /**< Resource path on the endpoint */
+    char ep[NANOCOAP_URI_MAX];     /**< Name of endpoint at RD server */
+    char base[NANOCOAP_URI_MAX];   /**< Registration base URI */
+    cord_lc_rd_t *rd;               /**< RD server the endpoint registered at */
+} cord_lc_ep_t;
+
+/**
+ * @brief  Information about a resource registered at a RD server
+ */
+typedef struct {
+    char path[NANOCOAP_URI_MAX];      /**< Resource path on the endpoint */
+    char rt[CORD_LC_ATTR_LEN];        /**< Application specific resource type */
+    char ifdesc[CORD_LC_ATTR_LEN];    /**< Interface description */
+    char anchor[NANOCOAP_URI_MAX];    /**< Anchor */
+} cord_lc_res_t;
+
+/**
+ * @brief Discover the lookup interface resources of the target RD server
+ *
+ * @param[out] rd       information about RD server at @p remote
+ *
+ * @return CORD_LC_OK on success
+ * @return CORD_LC_TIMEOUT if the discovery request times out
+ * @return CORD_LC_NORSC if one of the lookup interface for {res, ep, gp}
+ *                       resource type not found.
+ *                       Result **can** still be used
+ * @return CORD_LC_ERR on any other internal error
+ */
+int cord_lc_rd_init(cord_lc_rd_t *rd, const sock_udp_ep_t *remote);
+
+/**
+ * @brief Raw lookup for registered resources at a RD server
+ *
+ * @param[in]  rd               the target RD server
+ * @param[in]  content_format   data format wanted from RD server
+ * @param[out] result           buffer holding the raw response
+ * @param[in]  maxlen           max length of @p result
+ *
+ * @return length of result on success
+ * @return CORD_LC_NORSC if there is no resource lookup interface at @p rd
+ * @return CORD_LC_TIMEOUT if lookup timed out
+ * @return CORD_LC_ERR on any other internal error
+ */
+ssize_t cord_lc_res_raw(const cord_lc_rd_t *rd, uint8_t content_format,
```

Comment:
An array of `clif_param_t` parameter structs from #11189 could be used as the input filter.

Reply:
Yup, lots of the code in this PR can be replaced by clif e.g. clif_t instead of cord_lc_res_t and cord_lc_ep_t as both of them is just links actually. I think there's a lot more than can replaced.